sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import {
+  Application,
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+  getUserFullNameLikeSearch,
+  transformToEntitySortField,
+} from "@sims/sims-db";
+import { Brackets, Repository } from "typeorm";
+import { PaginatedResults, PaginationOptions } from "../../utilities";
+
+@Injectable()
+export class ApplicationOfferingChangeRequestService {
+  constructor(
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+  ) {}
+
+  /**
+   * Gets all eligible applications that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible applications that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
```

Comment:
The `records` is not adding any value here IMO.
I would prefer to have it named `getEligibleApplicationOfferingChanges`.

Reply:
`getEligibleApplicationOfferingChanges` makes sense to me.
When appending `Records` I was just concerned that it can become a pattern and we would start to see `Records` appended everywhere 😉